base.go

```diff
@@ -0,0 +1,75 @@
+package rpc
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+type base struct {

```

Comment:
Golang doesn't support OOP directly. We can mimic the inheritance using "embedding". Using the type named "base" seems awkward since there is no "inheritance." How about jsonRPCClient, jsonRPCCaller, jsonRPCRequester or something?

Reply:
I renamed this to `jsonRPCClient`